src/uu/hashsum/src/digest.rs

```diff
@@ -158,3 +170,62 @@ impl_digest_sha!(sha3::Sha3_384, 384);
 impl_digest_sha!(sha3::Sha3_512, 512);
 impl_digest_shake!(sha3::Shake128);
 impl_digest_shake!(sha3::Shake256);
+
+/// A struct that writes to a digest.
+///
+/// This struct wraps a [`Digest`] and provides a [`Write`]
+/// implementation that passes input bytes directly to the
+/// [`Digest::input`].
+///
+/// On Windows, the [`write`] implementation replaces instances of
+/// "\r\n" with "\n" before passing the input bytes to the [`digest`].
+pub struct DigestWriter<'a> {
+    digest: &'a mut Box<dyn Digest>,
+    binary: bool,
```

Comment:
You can either remove `binary` for non-windows platforms with a `#[cfg(windows])`. Alternatively, you can ignore the lint by adding `#[allow(dead_code)]` in front of `binary`.

Reply:
I added the `#[allow(dead_code)]`. I'll leave it to someone else to do something with `#[cfg(windows)]`.